ocimotel/repoclient.go

```diff
@@ -30,18 +35,52 @@ func NewOciRepo(ref *ociMotelReference) (r OciRepo, err error) {
 	if ref.port != -1 {
 		port = fmt.Sprintf("%d", ref.port)
 	}
-	r.url = fmt.Sprintf("http://%s:%s", server, port)
-	queryURI := fmt.Sprintf("%s/v2/", r.url)
-	client := &http.Client{}
-	resp, err := client.Get(queryURI)
-	if err != nil {
-		return r, err
+
+	insecureSkipVerify := (sys.DockerInsecureSkipTLSVerify == types.OptionalBoolTrue)
+	tlsClientConfig := &tls.Config{
+		InsecureSkipVerify: insecureSkipVerify,
 	}
-	defer resp.Body.Close()
-	if resp.StatusCode == 200 {
-		return r, nil
+
+	if sys.DockerCertPath != "" {
+		if err := tlsclientconfig.SetupCertificates(sys.DockerCertPath, tlsClientConfig); err != nil {
+			return r, err
+		}
 	}
-	return r, fmt.Errorf("Unexpected return code %d from %s", resp.StatusCode, r.url)
+	transport := &http.Transport{TLSClientConfig: tlsClientConfig}
+	client := &http.Client{Transport: transport}
+	creds := ""
+	if sys.DockerAuthConfig != nil {
+		a := sys.DockerAuthConfig
+		creds = base64.StdEncoding.EncodeToString([]byte(a.Username + ":" + a.Password))
+	}
+	r = OciRepo{ref: ref, authCreds: creds, client: client}
+
+	ping := func(scheme string) error {
+		r.url = fmt.Sprintf("%s://%s:%s", scheme, server, port)
+		queryURI := fmt.Sprintf("%s/v2/", r.url)
```

Comment:
Seems like we might want a url encode call here somewhere.

Reply:
> Seems like we might want a url encode call here somewhere.

Tried a cleanup patch, but not sure what this would buy us (except validation perhaps) because http client requests take URLs as strings which is the bulk of the business logic anyways.